src/components/UIControls.js

```diff
@@ -6,21 +6,20 @@ import { Subject } from 'rxjs';
 export default React.createClass({
 
   propTypes: {
-    navigate:   React.PropTypes.func.isRequired,
-    directions: React.PropTypes.array.isRequired
+    navigator: React.PropTypes.object.isRequired
   },
 
   isValidMotion: function (motion) {
-    return this.motions.indexOf(motion) !== -1;
+    return this.directions.indexOf(motion) !== -1;
   },
 
   componentWillMount: function () {
-    this.motions = Object.keys(this.props.motions);
-    this.clicks = new Subject(),
+    this.directions = this.props.navigator.getDirectionNames();
```

Comment:
While you're at it, if someone calls `isValidMotion` before `componentWillMount` it will crash saying `this.directions` isn't defined. We should use `this.props.navigator.getDirectionNames();` instead